### DISCUSSION TITLE
How do I customize file paths of snapshot images?

### DISCUSSION THREAD

**Discussion by yukonblue** [pointfreeco/swift-snapshot-testing#876] [Q&A] [answered]
I'm fairly new with using this library. While it's super handy dandy, one problem I have so far is that all the recorded snapshot images are named `MyViewController.1.png`, `MyViewController.2.png`, ..., etc.

This makes the snapshot images kind of useless as source artifacts because one cannot easily relate anyone to a set of UI traits. For example if I want to enumerate through combinations of light/dark mode and size class and generate all images, I wish the files are named something like `MyViewController.light.medium.png`, where I can easily tell which trait set it's referring to.

So far I haven't been able to figure out how to achieve this easily through the public APIs. Some guidance can be appreciated.

**Comment by mbrandonw** [comment, marked as answer]
Hi @yukonblue, you can provide a custom name for the snapshots by providing the `named` argument to `assertSnapshot`:

https://github.com/pointfreeco/swift-snapshot-testing/blob/2d0d95e5c5317d0f129fdca4255931c09a457652/Sources/SnapshotTesting/AssertSnapshot.swift#L61

However, it will be on you to make sure you have unique names for your snapshots.

**Reply by yukonblue** [reply]
> Hi @yukonblue, you can provide a custom name for the snapshots by providing the `named` argument to `assertSnapshot`:
> 
> https://github.com/pointfreeco/swift-snapshot-testing/blob/2d0d95e5c5317d0f129fdca4255931c09a457652/Sources/SnapshotTesting/AssertSnapshot.swift#L61
> 
> However, it will be on you to make sure you have unique names for your snapshots.

doh! Silly me for not spotting that earlier. Thanks a bunch @mbrandonw! 🙏
